drivers/at86rf2xx/at86rf2xx_netdev.c

```diff
@@ -166,15 +166,17 @@ static int _recv(netdev_t *netdev, void *buf, size_t len, void *info)
     (void)tmp;
 
     if (info != NULL) {
+        uint8_t rssi = 0;
         netdev_ieee802154_rx_info_t *radio_info = info;
         at86rf2xx_fb_read(dev, &(radio_info->lqi), 1);
 #ifndef MODULE_AT86RF231
-        at86rf2xx_fb_read(dev, &(radio_info->rssi), 1);
+        at86rf2xx_fb_read(dev, &(rssi), 1);
         at86rf2xx_fb_stop(dev);
 #else
         at86rf2xx_fb_stop(dev);
-        radio_info->rssi = at86rf2xx_reg_read(dev, AT86RF2XX_REG__PHY_ED_LEVEL);
+        rssi = at86rf2xx_reg_read(dev, AT86RF2XX_REG__PHY_ED_LEVEL);
 #endif
+        radio_info->rssi = -RSSI_BASE_VAL - rssi;
```

Comment:
@bergzand Please help me out, what is the reason for changing the algebraic sign of this calculation? Especially as you are changing `.rssi` to `int8_t` ...

Reply:
I forgot to change `.rssi` back to the original `uint8_t`. The sign here is flipped to have the value from the datasheet directly as a define. The sign is flipped to correct the algabraic formula from sec 8.5.3 from the datasheet of the at86rf233 to get a positive value.